src/tools/compiletest/src/runtest.rs

```diff
@@ -268,6 +274,7 @@ struct TestCx<'test> {
     props: &'test TestProps,
     testpaths: &'test TestPaths,
     revision: Option<&'test str>,
+    is_aux: bool,
```

Comment:
What is this thing and why does it change the pass mode?

Reply:
The `if self.is_aux { self.props.pass_mode } else { self.effective_pass_mode() };` line looks very suspicious.

Reply:
iirc something weird was going on with aux-build that caused failures with `--pass check` but I cannot seem to reproduce so it's probable that some other change made them go away.